packages/dataviews/src/single-selection-checkbox.js

```diff
@@ -32,10 +32,14 @@ export default function SingleSelectionCheckbox( {
 		<CheckboxControl
 			className="dataviews-view-table-selection-checkbox"
 			__nextHasNoMarginBottom
-			checked={ isSelected }
 			label={ selectionLabel }
-			disabled={ disabled }
+			aria-disabled={ disabled }
+			checked={ isSelected }
 			onChange={ () => {
+				if ( disabled ) {
+					return;
+				}
```

Comment:
Did we consider whether this behavior make sense for all usage of `CheckboxControl` and whether we should bake it in for any usage of the `disabled` prop? (Obviously not a blocker)

Reply:
When the checkbox is disabled, click or keyboard actions shouldn't trigger an actual change.

Reply:
Currently when a checkbox is `aria-disabled`, click and keyboard interactions will toggle the checkmark (visually at least). I don't think you'd ever want that, so it makes sense to add this to the component some how?

Reply:
My thinking was more: automatically use `aria-disabled` and prevent `onChange` when  `disabled` is passed. Like we do when `__experimentalIsFocusable` is true in buttons.

Reply:
I added an issue for this at #59411. Ideally we would address it in the CheckboxControl component first, but if we don't have time I'm fine with cleaning up later.

As for the default focusability of a disabled component, I'm kind of [leaning toward](https://github.com/WordPress/gutenberg/issues/56547#issuecomment-1894731461) sticking to the standard browser behavior (which is to not be focusable). Of course we would also provide an opt-in prop to maintain focusability, as in Button.